types/aws-lambda/trigger/appsync-resolver.d.ts

```diff
@@ -0,0 +1,44 @@
+import { Handler } from '../handler';
+
+export type AppSyncResolverHandler<T, V> = Handler<AppSyncResolverEvent<T>, V | V[]>;
+
+/**
+ * See https://docs.aws.amazon.com/appsync/latest/devguide/resolver-context-reference.html
+ *
+ * @param T type of the arguments
+ */
+export interface AppSyncResolverEvent<T> {
+    arguments: T;
+    identity?: AppSyncIdentityIAM | AppSyncIdentityCognito;
+    source?: { [key: string]: any };
+    request: {
+        headers: { [key: string]: string };
+    };
+    info: {
+        selectionSetList: string[];
+        selectionSetGraphQL: string;
+        parentTypeName: string;
+        fieldName: string;
+        variables: { [key: string]: any };
+    };
+}
+
+export interface AppSyncIdentityIAM {
+    accountId: string;
+    cognitoIdentityPoolId: string;
+    cognitoIdentityId: string;
+    sourceIp: string[];
+    username: string;
+    userArn: string;
+    cognitoIdentityAuthType: string;
```

Comment:
This should probably be `"authenticated" | "unauthenticated"`? Maybe not, if AWS adds more types, but I don't think they can.